test/splitter.js

```diff
@@ -59,9 +64,98 @@ contract('Splitter', accounts => {
     it('should not split the ether to Bob and Carol', async () => {
       // Act & Assert
       await truffleAssert.reverts(
-        contract.methods.split().send({from: notAlice, value: 2}),
+        contract.methods.split().send({from: someoneElse, value: 2}),
         'Only Alice can split!'
       );
     });
   });
+
+  context('When Bob withdraws', () => {
+    let bobBeforeWithdrawBalance;
+    beforeEach(async () => {
+      // Arrange
+      bobBeforeWithdrawBalance = new BN(await web3.eth.getBalance(bob));
+      await contract.methods.split().send({from: alice, value: web3.utils.toWei('0.06', 'ether')});
+
+      // Act
+      await contract.methods.withdraw().send({from: bob});
+    });
+
+    it('should withdraw the ether to Bob\'s account', async () => {
+      // Assert
+      assert.ok(new BN(await web3.eth.getBalance(bob)).gt(bobBeforeWithdrawBalance), 'balance should increase after withdraw');
+    });
+
+    it('Bob\'s balance in the contract should set to zero', async () => {
+      // Assert
+      assert.equal((await contract.methods.bobBalance().call()), 0);
+    });
+
+    it('the contract balance should decrease 0.03 ether', async () => {
+      // Assert
+      assert.equal((await web3.eth.getBalance(contract.options.address)), web3.utils.toWei('0.03', 'ether'));
+    });
```

Comment:
And is it Bob indeed who got the money?

Reply:
Shall I add `assert.equal((await contract.methods.carolBalance().call()), web3.utils.toWei('0.03', 'ether'));` here?

Reply:
No, when I say "Bob got the money", I am not saying "the contract says it owes 0 to Bob". I am saying that really, the money is now in control of Bob's private key.
How do you check that?

Reply:
Has https://github.com/JakeLin/splitter/pull/1/files#diff-c82206304e3a5309e62cc5aa28422d87R84 checked Bob gets the money or not? Because Bob needs to spend some gas, it's difficult to check Bob gets exact 0.03 ether from the contract to his own account.

Reply:
> Because Bob needs to spend some gas, it's difficult

That's exactly what we want you to do. Did you do the gas challenge?

Reply:
BTW, the link leads me to this very page, not sure what you are pointing to.

Reply:
thanks, I did the gas challenge for sending ether, but not call a contract. I was pointing to

```
it('should withdraw the ether to Bob\'s account', async () => {
      // Assert
      assert.ok(new BN(await web3.eth.getBalance(bob)).gt(bobBeforeWithdrawBalance), 'balance should increase after withdraw');
    });
```

Reply:
@xavierlepretre I probably need some help here, do I need to specify the gas every time and  I will get the same balance every time after calling `withdraw` method? Thanks

Reply:
You need to calculate the gas cost after the withdrawal. How do you calculate the gas cost?